class.jetpack-jitm.php

```diff
@@ -0,0 +1,141 @@
+			echo '</a></p></div>';
+
+			// Enqueue javascript to handle jitm notice events
+			wp_enqueue_script( 'jetpack-jitm-js', plugins_url( '_inc/jetpack-jitm.js', JETPACK__PLUGIN_FILE ),
+				array( 'jquery' ), JETPACK__VERSION . '-20121111' );
+			wp_localize_script(
+				'jetpack-jitm-js',
+				'jitmL10n',
+				array(
+					'jumpstart_stats_urls'              => $this->build_jitm_stats_urls( array( 'dismiss', 'learnmore' ) ),
+					'jumpstart_plugin_serp_stats_urls'  => $this->build_jitm_stats_urls( array( 'dismiss', 'learnmore' ), '-plugin-serp'),
+					'hide_jitm_plugins'                 => $hide_jitm,
+					'ajaxurl'                           => admin_url( 'admin-ajax.php' ),
+					'jitm_nonce'                        => wp_create_nonce( 'jetpack-jitm-nonce' ),
+				)
+			);
+
+			//JITM is being viewed send data to MC Stats
+			$search_term_slug = str_replace( ' ', '-', $search_term);
+			$jetpack = Jetpack::init();
+			$jetpack->stat( 'jitm', 'viewed' );
+			$jetpack->stat( 'jitm-plugin-serp', 'viewed,'.$module_info[0]['module_slug'].',search:-'.$search_term_slug );
+			$jetpack->do_stats( 'server_side' );
+		}
+	}
+
+	/*
+	 * Build an array of a specific module tag.
+	 *
+	 * @param  string Name of the module tag
+	 * @return array  The module slug, config url, and name of each Jump Start module
+	 */
+	function jitm_module_tags( $search_term ) {
+		$modules = Jetpack_Admin::init()->get_modules();
+
+		$module_info = array();
+
+		foreach ( $modules as $module => $value ) {
+			if ( ! empty( $value['jitm_tags'] ) && in_array( $search_term, $value['jitm_tags'] ) ) {
+				$module_info[] = array(
+					'module_slug'   => $value['module'],
+					'module_name'   => $value['name'],
+				);
+
+				//we only need one result, if we find it move on
+				break;
+
+			}
+		}
+		return $module_info;
+	}
+
+	/*
+	* Build an array of JITM stats urls.
+	* requires the build URL args passed as an array
+	*
+	* @param array $jitm_stats
+	* @return (array) of built stats urls
+	*/
+	function build_jitm_stats_urls( $jitm_stats, $jitm_stats_suffix = false ) {
+		$jitm_urls = array();
+		$jitm_stats_report = $jitm_stats_suffix ? 'x_jetpack-jitm'.$jitm_stats_suffix : 'x_jetpack-jitm';
+
+		foreach ( $jitm_stats as $value) {
+			$jitm_urls[$value] = Jetpack::build_stats_url( array( $jitm_stats_report => $value ) );
+		}
+
+		return $jitm_urls;
+
+	}
+
+	/*
+	* Function to enqueue jitm css specifically in the header
+	*/
+	function enqueue_css_header( $hook ) {
+
+		$wp_styles = new WP_Styles();
+
+		if ( ! isset( $_GET['tab'] ) || 'search' != $_GET['tab'] && 'plugin-install.php' != $hook ) {
+			return;
+		}
+		$min = ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ? '' : '.min';
+
+		wp_enqueue_style( 'jetpack-jitm-css', plugins_url( "css/jetpack-admin-jitm{$min}.css", JETPACK__PLUGIN_FILE ), false, JETPACK__VERSION . '-20121016' );
+		$wp_styles->add_data( 'jetpack-jitm-css', 'rtl', true );
+	}
+
+}
+
+if ( apply_filters( 'Jetpack_JITM_plugin_search', false ) ) {
```

Comment:
Reposting here so it doesn't get lost.
1. Inline docs
2. What about `jetpack_jitm_plugin_search` to keep filters all lowercase? I don't think we have a mixed-case filter name elsewhere, I don't think.